application-workloads/visualstudio/az-400-dev-env/azuredeploy.json

```diff
@@ -0,0 +1,263 @@
+            "apiVersion": "2020-04-01",
+            "name": "vslab-nic",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses', 'vslab-PublicIP')]",
+                "[resourceId('Microsoft.Network/virtualNetworks', 'vslab-VirtualNetwork')]"
+            ],
+            "tags": {
+                "displayName": "VSLab Network Interface"
+            },
+            "properties": {
+                "ipConfigurations": [
+                    {
+                        "name": "ipConfig1",
+                        "properties": {
+                            "privateIPAllocationMethod": "Dynamic",
+                            "publicIPAddress": {
+                                "id": "[resourceId('Microsoft.Network/publicIPAddresses', 'vslab-PublicIP')]"
+                            },
+                            "subnet": {
+                                "id": "[resourceId('Microsoft.Network/virtualNetworks/subnets', 'vslab-VirtualNetwork', 'vslab-vnet-subnet')]"
+                            }
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2019-07-01",
+            "name": "vsLab",
+            "location": "[parameters('location')]",
+            "dependsOn": [ 
+                "[resourceId('Microsoft.Storage/storageAccounts', parameters('storagename'))]",
+                "[resourceId('Microsoft.Network/networkInterfaces', 'vslab-nic')]"
+            ],
+            "tags": {
+                "displayName": "VS2019 Lab VM"
+            },
+            "properties": {
+                "hardwareProfile": {
+                    "vmSize": "[parameters('vmsize')]"
+                },
+                "osProfile": {
+                    "computerName": "LabVM",
+                    "adminUsername": "[parameters('adminUser')]",
+                    "adminPassword": "[parameters('adminPassword')]"
+                },
+                "storageProfile": {
+                    "imageReference": {
+                        "publisher": "MicrosoftVisualStudio",
+                        "offer": "VisualStudio",
+                        "sku": "VS-2017-Comm-Latest-WS2016",
+                        "version": "latest"
+                    },
+                    "osDisk": {
+                        "name": "LabVMOSDisk",
+                        "caching": "ReadWrite",
+                        "createOption": "FromImage"
+                    }
+                },
+                "networkProfile": {
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', 'vslab-nic')]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
```

Comment:
is this really needed for the lab scenario?

Reply:
Not really required for lab ! 
I have now removed it